src/quicktube.js

```diff
@@ -0,0 +1,284 @@
+        }, options);
+
+        // Concatenate in a more readable way :D
+        this._settings = `?autoplay=1\
+&showinfo=${this.options.showInfo}\
+&autohide=${this.options.autohide}\
+&color=${this.options.color}\
+&enablejsapi=1\
+&playerapiid=ytplayer\
+&wmode=${this.options.wmode}`;
+
+        this.iframeClass = 'quicktube__iframe';
+        this.videoId = videoId;
+        this.videoEl = document.querySelector(`[data-quicktube="${videoId}"]`);
+        this.poster = this.videoEl.querySelector('[data-quicktube-poster]');
+        this.isMobileSafari = isMobileSafari();
+        this.onClick = this.onClick.bind(this);
+        this.stopVideo = this.stopVideo.bind(this);
+        this.onPlayerReady = this.onPlayerReady.bind(this);
+        this.onPlayerStateChange = this.onPlayerStateChange.bind(this);
+        this.onPlayerError = this.onPlayerError.bind(this);
+
+        const playButton = document.querySelector(`[data-quicktube-play="${videoId}"]`);
+        const stopButton = document.querySelector(`[data-quicktube-stop="${videoId}"]`);
+
+        playButton.addEventListener('click', this.onClick, false);
+
+        playButton.addEventListener('keydown', (e) => {
+            if (e.keyCode === KEY_CODES.enter) {
+                this.onClick();
+            }
+        }, false);
+
+        stopButton.addEventListener('click', () => {
+            this.stopVideo();
+        }, false);
+    }
+
+    onClick() {
+        const iframeContainer = this.videoEl.querySelector('[data-quicktube-video]');
+        const videoIframes = iframeContainer.getElementsByTagName('iframe');
+
+        // defines whether video has already been loaded and you want to play again
+        let iframe = false;
+        if (videoIframes.length > 0) {
+            iframe = videoIframes[0];
+        }
+
+        if (!iframe) {
+            iframe = this.createIframePlayer();
+            iframeContainer.appendChild(iframe);
+            this.quicktubePlayer = new YT.Player(this.videoId, {
+                events: {
+                    onStateChange: this.onPlayerStateChange,
+                    onReady: this.onPlayerReady,
+                    onError: this.onPlayerError,
+                },
+            });
+            YT.gaLastAction = 'p';
+        }
+
+        // Only trigger force video play if not Mobile safari as playVideo function not supported
+        if (!this.isMobileSafari) {
+            if (this.quicktubePlayer.playVideo) {
+                this.quicktubePlayer.playVideo();
+            }
+        }
+
+        // Check if video isn't already playing
+        if (!this.videoEl.getAttribute('data-video-playing')) {
+            this.hidePosterFrame();
+            this.addActiveState();
+            window.dispatchEvent(new Event('quicktube:play'));
+        }
+    }
+
+    addActiveState() {
+        this.videoEl.classList.add(this.options.activeClass);
+        this.videoEl.classList.remove(this.options.pausedClass);
```

Comment:
Yes, unsure exactly what to do about this. Since I imagine the youtube video block has it's own accessibility traits so not sure what we are required to do just yet. I shall investigate properly